salesagility/SuiteCRM
# Feature/robo api

## Description



Created a robo task for setting up the V8 API. Also added CliRunnerTrait.php for accessing a db connection from CLI.

## How To Test This

1. Run ```composer install```.
2. Run ```./vendor/bin/robo api:configure-v8```.
3. Check that the ```Api/V8/OAuth2/``` private/public keys have been generated and given the correct permissions.
4. Check that the ```Api/Core/Config/ApiConfig.php``` constant ```OAUTH2_ENCRYPTION_KEY``` has been updated.
5. Check that the .htacess file has been rebuilt.
6. Check that the CLI output is correct.

Check that all the following tasks work:
```
  api:configure-v8           Configures the SuiteCRM V8 API with all defaults
  api:create-client          Creates OAuth2 client
  api:create-user            Creates a SuiteCRM user for the V8 API
  api:export-postman-env     Export a postman environment for the V8 API
  api:generate-keys          Generate OAuth2 public/private keys
  api:rebuild-htaccess-file  Rebuild .Htaccess file
  api:set-key-permissions    Sets the Oauth2 key permissions
  api:update-encryption-key  Update OAuth2 encryption keys
```

## Types of changes


### Final checklist
